README.md

```diff
@@ -0,0 +1,142 @@
+Servo's implementation. We don't want to limit the API too early for the sake
+of simplification (this can be done later on top of this API).
+
+This low level API covers only **Servo <-> embedder** communication. This is
+not enough to build a full browser application, which requires some other extra
+powers, like access to the operating system. This project doesn't address this
+problematic as it's an orthogonal problem (we usually refer to app level or
+operating system access as "Runtime").
+
+# Overview
+
+Basic structure is: A browser holds a reference to history entries, that hold
+a reference to a pipeline. A Browser is rendered in a Viewport.
+A Pipeline can be rendered independently.
+
+Pipeline can live without a history entry or a browser (orphan pipeline).
+
+The embedder holds a list of Browser objects.
+
+## [Browser](webidls/Browser.webidl)
+
+Servo equivalent: a top level Frame.
+
+Equivalent of a tab, a top level browsing context. Holds a reference to a
+StorageSession object ([see bellow](#storagesession)). Holds a list of HistoryEntries
+([see below](#historyentry)). Holds a list of ContentBlockers ([see below](#contentblockers)).
+One history entry is active (current pipeline).
+
+Holds default properties for future pipelines.
+
+Responsible for the navigation through the history. Notify when a new entry
+is available (user clicks on a link).
+
+## [HistoryEntry](webidls/HistoryEntry.webidl)
+
+Servo equivalent: FrameState.
+
+Holds an optional reference to a pipeline. Pipeline might be dead. Holds a
+reference to LoadData ([see below](#loaddata)).
+
+## [Pipeline](webidls/Pipeline.webidl)
+
+A page.
+
+A reference to a servo pipeline. Offers access to many internal information
+about the document. Fires many events to keep track of the document status
+(loading state, security state, etc) and request user actions (prompts,
+security questions, …).
+
+Can be pending, loading, interactive, complete (loaded).
+
+A pipeline implements some extra interfaces: [Editable](webidls/Editable.webidl)
+(to build the app "edit" menu), [FindInPage](webidls/FindInPage.webidl) (to build
+a in page text search), [HTTPObserverManager](webidls/HTTPObserverManager.webidl)
+(to track and overwrite HTTP connections), [MultimediaManager](webidls/MultimediaManager.webidl)
+(to track multimedia content, to silence a tab for example),
+[Printable](webidls/Printable.webidl) (to implement Print to printer or Print to PDF).
+
+To preloada page (first result in the urlbar for example),
+it's possible to create a new "orphan" pipeline and later attach it to a
+Browser, as long as they share the same session.
+
+## [LoadData](webidls/LoadData.webidl)
+
+Serializable. Minimal set of information to create or restore a pipeline. Is
+used to save a history entry if pipeline is being purged (#11893). Is used to
+restore session (list of LoadData can be store on disk). Is used to transmit
+request, to the embedder, to open a new window or a new tab.
+
```

Comment:
I found this to be very confusing, as it took me several passes to actually get it. I think rephrasing would help. Maybe something like: information passed to a handler when request to open new window / tab is created.